presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -386,20 +404,128 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
             }
             ColumnStatistics.Builder columnStatistics = ColumnStatistics.builder();
             RangeColumnStatistics.Builder rangeStatistics = RangeColumnStatistics.builder();
+
+            List<Object> lowValueCandidates = ImmutableList.of();
+            List<Object> highValueCandidates = ImmutableList.of();
+
+            Type prestoType = typeManager.getType(hiveColumnHandle.getTypeSignature());
             if (hiveColumnHandle.isPartitionKey()) {
                 rangeStatistics.setDistinctValuesCount(countDistinctPartitionKeys(hiveColumnHandle, hivePartitions));
                 rangeStatistics.setNullsFraction(calculateNullsFractionForPartitioningKey(hiveColumnHandle, hivePartitions, partitionStatistics));
+                if (isLowHighSupportedForType(prestoType)) {
+                    lowValueCandidates = hivePartitions.stream()
+                            .map(HivePartition::getKeys)
+                            .map(keys -> keys.get(hiveColumnHandle))
+                            .filter(value -> !value.isNull())
+                            .map(NullableValue::getValue)
+                            .distinct()
+                            .collect(toImmutableList());
+                    highValueCandidates = lowValueCandidates;
+                }
             }
             else {
                 rangeStatistics.setDistinctValuesCount(calculateDistinctValuesCount(partitionStatistics, columnName));
                 rangeStatistics.setNullsFraction(calculateNullsFraction(partitionStatistics, columnName, rowCount));
+
+                if (isLowHighSupportedForType(prestoType)) {
+                    lowValueCandidates = partitionStatistics.values().stream()
```

Comment:
maybe add `TODO` to better model cases where some partitions are unanalyzed. One can imaging that this few unanalyzed partitions might change min/max value dramatically causing underestimating final stats